### PR TITLE
Error handling around artifact upload

### PR DESCRIPTION
**Pull request by jared-verta**
## Impact and Context

- Adds logging around artifact upload failure

### CODE REVIEW

File: deployment/promotion/promote.py [hunk 1]
```diff
@@ -183,12 +182,24 @@ def upload_artifact(auth, model_version_id, artifact):
     key = artifact['key']
     print("Uploading artifact '%s'" % key)
 
-    upload_url = get_artifact_url(auth, model_version_id, artifact)
+    put_url = get_artifact_url(auth, model_version_id, artifact)
```
Comment (hmacdonald-verta) [inline deployment/promotion/promote.py:185]: confuzzled why you're calling it `put_url` when it's clearly a get

Comment (jared-verta) [reply deployment/promotion/promote.py:185]: It is used below in the call to `requests.put`, the `get_artifact_url` doesn't refer to the HTTP verb, but that we are fetching an S3 pre-signed URL from the Deploy API

Comment (hmacdonald-verta) [reply deployment/promotion/promote.py:185]: well it's obviously not a get, it's a put (line 187)

Comment (jared-verta) [reply deployment/promotion/promote.py:185]: (`artifact` here is really a map which contains `method` and `key`)

Comment (jared-verta) [reply deployment/promotion/promote.py]: Renamed to `signed_artifact_url`

**Review by hmacdonald-verta — APPROVED** [review]

**Pull request merged**